packages/containers/src/DeleteResource/deleteResource.sagas.js

```diff
@@ -71,21 +71,21 @@ export function* deleteResourceValidate(uri, resourceType, itemId, resourcePath)
 /**
  * Return a saga for delete resource confirmation dialog box.
  * Race between cancel and confirm deleting the resource.
- * @param {string} uri
- * @param {string} resourceType
- * @param {Array<String>} [resourcePath]
+ * @param {object} sagaParams the params to setup the saga
+ * @param {string} sagaParams.uri the uri of the endpoint to make the calls
+ * @param {string} sagaParams.resourceType the resource type to call delete endpoint ( at least )
+ * @param {Array<String>} sagaParams.resourcePath
+ * @param {string} sagaParams.redirectUrl url to redirect after delete action is done or cancel
  */
-export default function deleteResource(uri, resourceType, resourcePath) {
-	return function* deleteResourceSaga() {
-		const { redirectUrl, model } = yield take(deleteResourceConst.DIALOG_BOX_DELETE_RESOURCE);
-		const { id } = model;
+export default function deleteResource({ uri, resourceType, resourcePath, redirectUrl }) {
+	return function* deleteResourceSaga(routerParams) {
 		try {
 			yield race({
 				deleteConfirmationValidate: call(
 					deleteResourceValidate,
 					uri,
 					resourceType,
-					id,
+					routerParams.id,
```

Comment:
problem i see here is how to handle routes where the the resource indentifier is not strictly called `id`, how to deal with nested routes if we force all resources identifier to be called `id`

example `connection/:id/dataset/:id` with a delete at the end ?